### PR TITLE
Fix private urls using colon separator

### PR DESCRIPTION
**Pull request by Apercu**
People were still experiencing issues with private urls using colons as separator even after the fix introduced in #2384.

This fix basically replace the initial pathname part of the parsed url in the git resolver, and reformat exploded parts onto the equivalent url, removing the colon culprit.

Unsure about why the require onto the `url` module is done the old way, I've decided to follow consistency, but can refactor if needed. Related issues referenced in commit.

:wave: @chrisirhc @lxe

### CODE REVIEW

**Comment by chrisirhc** [comment]
The might not work for some git urls since `:` doesn't always translate to `/`. The conversion needs to be through git/ssh client which understand in their own way what the `:` means based on the `.gitconfig` or `.ssh/config`. I've mentioned one way of solving this is to not attempt to rewrite the URL (since it'll just fail) and just send it straight to the git client for handling.

File: __tests__/resolvers/exotics/git-resolver.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    'package@git@bitbucket.org:team/repo.git',
+  ];
+
+  urls.forEach((url) => {
+    expect(GitResolver.transformUrl(parse(url))).toBe(url);
+  });
+
+});
+
+test('GitResolver transformUrl affect host colon separated urls', () => {
```
Comment (btford) [inline __tests__/resolvers/exotics/git-resolver.js]: Maybe call this `'GitResolver transformUrl replaces the colon on colon separated urls'` ?

File: src/resolvers/exotics/git-resolver.js [hunk 2]
```diff
@@ -60,6 +61,14 @@ export default class GitResolver extends ExoticResolver {
     return false;
   }
 
+  // This transformUrl util is here to replace colon separators in the pathname
+  // from private urls. It takes the url parts retrieved using urlFormat and
+  // returns the associated url. Related to #573, introduced in #2519.
+  static transformUrl(parts) : string {
```
Comment (btford) [inline src/resolvers/exotics/git-resolver.js]: I think we can probably come up with a more descriptive name than "transform."

**Comment by bestander** [comment]
Nice work, thanks a lot for the tests!

**Pull request merged**